smart-connector-rest-server/src/main/resources/openapi-sc.yaml

```diff
@@ -678,14 +678,18 @@ components:
             type: string
     AskResult:
       type: object
-      required: [bindingSet, exchangeInfo]
+      required: [bindingSet, exchangeInfo, knowledgeGaps]
```

Comment:
Should we make the knowledge gaps optional? I.e. remove them from the required attribute list? In some scenarios (like knowledge gaps configured to be turned off or there not being any knowledge gaps) it might not be required to have the (possibly empty) knowledge gaps list. Not sure, though 😄

Reply:
@bnouwt : I think it would be good when the Rest server Ask request gets a boolean flag named `KnowledgeGapsEnabled` that indicates whether or not the user wants to get knowledge gap information. Similar to the `reasonerEnabled` flag when adding a KB. Agreed?

Reply:
Not sure yet what we should go for.

I am also wondering whether this configuration option should be configured at Knowledge Interaction (KI) *activation* (so when calling the `POST /sc/ask` operation) or at KI *registration* (so when calling the `POST /sc/ki` operation). So, if it is something users would want to change each time they activate a KI or whether it is something that you know when registering a KI and it does not change between activations. I am leaning towards the latter, so that would mean configuring it when registering a KI. This option might, however, be a bit more tedious to implement.

Reply:
All right...then let's go for the latter option...provide the option when registering a KI...I'll come up with a proposal for that.

Reply:
@bnouwt, I implemented the `knowledgeGapsEnabled` flag in the RestAPI, which runs through the nested Ask knowledge interaction execution of the knowledge engine. In addition, I made 4 tests to check the behavior of the SmartConnector java API. See: [here](https://github.com/TNO/knowledge-engine/tree/rest_api_with_knowledge_gaps/smart-connector/src/test/java/eu/knowledge/engine/smartconnector/api). Next step is to make some RestAPI tests to check the behavior at the RestAPI.

Reply:
Hey @JackJackie, thanks! That looks very thorough 👍  I will take a closer look the coming days.

Regarding the REST API tests, it would indeed be best to add those, but they are also quite some effort, so it might not be worth it.